python/ucxx/ucxx/testing.py

```diff
@@ -6,6 +6,30 @@
 from typing import Type, Union
 
 
+def join_processes(
+    processes: list[Type[BaseProcess]],
+    timeout: Union[float, int],
+) -> None:
+    """
+    Join a list of processes with a combined timeout.
+
+    Join a list of processes with a combined timeout, for each process `join()`
+    is called with a timeout equal to the difference of `timeout` and the time
+    elapsed since this function was called.
+
+    Parameters
+    ----------
+    processes:
+        The list of processes to be joined.
+    timeout: float or integer
+        Maximum time to wait for all the processes to be joined.
+    """
+    start = time.monotonic()
+    for p in processes:
+        t = timeout - (time.monotonic() - start)
```

Comment:
Yes, I tested that and it behaves the same as `timeout=0`, essentially returns immediately, although the [documentation](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Process.join) only mentions positive timeouts (to me implies any non-positive number is indeed expected to return immediately), do you think we should attempt to make the value always non-negative in case this is not documented because it's undefined behavior/implementation-dependent?

Reply:
I think that's ok as is (I had a read through the code as well, and they internally do `max(0, timeout)` I think.